src/apps/testapps/testPolygon.c

```diff
@@ -553,4 +553,34 @@ SUITE(polygon) {
 
         H3_EXPORT(destroyLinkedPolygon)(&polygon);
     }
+
+    TEST(normalizeMultiPolygon_unassignedHole) {
+        GeoCoord verts[] = {{0, 0}, {0, 1}, {1, 1}, {1, 0}};
+
+        LinkedGeoLoop* outer = malloc(sizeof(*outer));
+        assert(outer != NULL);
+        createLinkedLoop(outer, &verts[0], 4);
+
+        GeoCoord verts2[] = {{2, 2}, {3, 3}, {2, 3}};
+
+        LinkedGeoLoop* inner = malloc(sizeof(*inner));
+        assert(inner != NULL);
+        createLinkedLoop(inner, &verts2[0], 3);
+
+        LinkedGeoPolygon polygon = {0};
+        addLinkedLoop(&polygon, inner);
+        addLinkedLoop(&polygon, outer);
+
+        int result = normalizeMultiPolygon(&polygon);
+
+        t_assert(result == NORMALIZATION_ERR_UNASSIGNED_HOLES,
+                 "Expected error code returned");
+
+        t_assert(countLinkedPolygons(&polygon) == 1, "Polygon count correct");
+        t_assert(countLinkedLoops(&polygon) == 1,
+                 "Loop count on first polygon correct");
+        t_assert(polygon.first == outer, "Got expected outer loop");
```

Comment:
Is it worth asserting output here? My assumption would be that if we are returning error codes, we relinquish all guarantees about output when you get one - so no behavior after an error code is returned should be assertable, because it's all undefined.

Reply:
It may not be, since the polygon doesn't have all the features anymore.